Orange/widgets/settings.py

```diff
@@ -1016,23 +1023,24 @@ def settings_to_widget(self, widget, domain, *args):
     def encode_variable(var):
         return var.name, 100 + vartype(var)
 
-    def encode_setting(self, context, setting, value):
+    @classmethod
```

Comment:
What is the reasoning behind changing this to a class method?

Reply:
Class methods are easier/cleaner to test because they don't require having an instance.  I actually test this method like this, https://github.com/biolab/orange3/pull/3697/files#diff-0528176a310e674da65e2227537d9a05R285.

I see no disadvantages; derived classes can still have a non-class method and I do not see having `cls` instead of `self` as a different signature. (Neither does pylint consider them as different.)

Besides, I locally use a stricter setting for pylint with just a few disabled warnings,

```
disable=
    missing-docstring,  # I guess not
    no-name-in-module, no-member  # too many false positives from Qt
    too-many-ancestors, # I don't think this is a problem
    no-else-return      # else's may actually improve readability
```

so both pylint and pycharm suggest turning this into a class method.

Reply:
Thanks for the explanation. I have nothing against the change, I was just curious.